packages/signature-controller/src/SignatureController.ts

```diff
@@ -807,13 +829,21 @@ export class SignatureController extends BaseController<
   async #requestApproval(
     msgParams: AbstractMessageParamsMetamask,
     type: ApprovalType,
+    { traceContext }: { traceContext?: TraceContext },
   ): Promise<AddResult> {
     const id = msgParams.metamaskId as string;
     const origin = msgParams.origin || ORIGIN_METAMASK;
 
     // We are explicitly cloning the message params here to prevent the mutation errors on development mode
     // Because sending it through the messaging system will make the object read only
     const clonedMsgParams = cloneDeep(msgParams);
+
+    await this.#trace({
+      name: 'Notification Display',
```

Comment:
For audit sake, we use an enum for these values in the extension but it would be and ongoing effort to synchronise them across the core and client repos using an enum. So likely a future concern.